astroplan/scheduling.py

```diff
@@ -124,8 +126,170 @@ def components(self, val):
 
         self._components = val
         self.duration = duration
+        
+    @classmethod
+    @u.quantity_input(duration=u.second)
+    def from_duration(cls, duration):
+        # for testing how to put transitions between observations during
+        # scheduling without considering the complexities of duration
+        tb = TransitionBlock({None: 0*u.second})
+        tb.duration = duration
+        return tb
+
+
+class Schedule(object):
+    """
+    An object that represents a astronomical schedule
+    
+    Parameters:
+    -----------
+    start_time : `~astropy.time.Time`
+        The starting time of the schedule; the start of your 
+        observing window
+    end_time : `~astropy.time.Time`
+        The ending time of the schedule; the end of your
+        observing window
+    constraints : sequence of `Constraint`s
+        these are constraints that apply to the entire schedule
+    """
+    # as currently written, there should be no consecutive unoccupied slots
+    # this should change to allow for more flexibility (e.g. dark slots, grey slots)
+    
+    def __init__(self, start_time, end_time, constraints=None):
+        self.slots = [Slot(start_time, end_time)]
+        self.constraints = constraints
+        self.slew_duration = 4*u.min
+        # TODO: replace/overwrite slew_duration with Transitioner calls
 
+    def __repr__(self):
+        # once TransitionBlocks are defined, this should be scheduled_blocks
+        # TODO: make a better repr
+        for block in self.observing_blocks:
+            if hasattr(block, 'target'):
+                try:
+                    print(block.target.name, 'starting @',
+                          block.start_time.iso, 'lasting', block.duration)
+                except:
+                    print(block.name, 'starting @', block.start_time.iso,
+                          'lasting', block.duration)
+            else:
+                print('trans starting @', block.start_time.iso, 'lasting',
+                      block.duration, ':', block.components)
+        return 'done'
```

Comment:
Before we get too into this, let's note here that the `__repr__` is not of paramount importance yet. 

This could be a touchy subject, but I think they should really be single line unless there's a really good reason that they shouldn't be. The point of a `__repr__` isn't to display all of the information within the object, it's to summarize it. Maybe you could make the `__repr__` simply print the object name (like most objects do) and the number of scheduled OB's?